### PR TITLE
Unnecessary Exception is thrown when pub sub configure node is empty,…

### PR DESCRIPTION
**Pull request by annovanvliet**
… for instance in case of an auth error.

### CODE REVIEW

File: smack-extensions/src/main/java/org/jivesoftware/smackx/pubsub/provider/FormNodeProvider.java [hunk 1]
```diff
@@ -35,6 +34,11 @@
 public class FormNodeProvider extends EmbeddedExtensionProvider<FormNode> {
     @Override
     protected FormNode createReturnExtension(String currentElement, String currentNamespace, Map<String, String> attributeMap, List<? extends ExtensionElement> content) {
-        return new FormNode(FormNodeType.valueOfFromElementName(currentElement, currentNamespace), attributeMap.get("node"), (DataForm) content.iterator().next());
+      
+        if ( !content.isEmpty() ) {
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/pubsub/provider/FormNodeProvider.java:38]: "If content is empty, then return null" is easier to read.

**Review by Flowdalic — CHANGES_REQUESTED** [review]
Can you share the stanza that triggers this? Also ideally add a test case for this.

**Comment by annovanvliet** [comment]
An example is the following returned from OF:
<iq type="error" id="6LXNC-48" from="pubsub.openfire.xmpp.test" to="anno@openfire.xmpp.test/5dsi4g084a">
  <pubsub xmlns="http://jabber.org/protocol/pubsub#owner">
    <configure node="fdp/submitted/spot_report"/>
  </pubsub>
  <error code="403" type="auth">
    <forbidden xmlns="urn:ietf:params:xml:ns:xmpp-stanzas"/>
  </error>
</iq>

**Comment by Flowdalic** [comment]
It appears to me the correct fix to this would be to allow `null` as `DataForm` argument to `FormNode`.

**Pull request closed without merging**